pkg/operator/sync.go

```diff
@@ -1569,13 +1569,14 @@ func (optr *Operator) syncRequiredMachineConfigPools(config *renderConfig, co *c
 		// This was needed in-case the cluster is doing a master pool update when a new MachineConfiguration was applied.
 		// This prevents the need to wait for all the master nodes (as the syncAll function of the operator will be
 		// "stuck" here in such a case) to update before the MachineConfiguration status is updated.
-		syncErr := optr.syncMachineConfiguration(config, co)
-		// Update the degrade condition if there was an error reported by syncMachineConfiguration
-		newCO := co.DeepCopy()
-		optr.syncDegradedStatus(newCO, syncError{task: "MachineConfiguration", err: syncErr})
-		co, syncErr = optr.updateClusterOperatorStatus(co, &newCO.Status, lastErr)
-		if syncErr != nil {
-			klog.Errorf("Error updating cluster operator status: %q", syncErr)
+		if syncErr := optr.syncMachineConfiguration(config, co); syncErr != nil {
```

Comment:
Makes sense. Just curious, if we were to fix the error, what is clearing the degrade? Did it used to be here (from nil error) or did something else cover that?

Reply:
This is what clears the degrade condition: https://github.com/openshift/machine-config-operator/blob/f9abd9d45af1028b03ba00fadf5053523a2d7e4f/pkg/operator/sync.go#L199 

We get "trapped" in `syncRequiredMachineConfigPools` during an update, so if a degrade happens in that window, it won't get cleared till all master nodes are done updating.